### ISSUE TITLE
Disable automatic store hydration (and do it manually)

### ISSUE THREAD

**Issue by dmitry** [quasarframework/quasar#5903]
After the #5795 (and #5853) when store hydration changed it's order and now running before calling the boot plugins our application have an issue, because we are using apollo as our `store` and it initialized only while the boot.

It's initialized in `store/index.js` like that:

```
export default ({ }) => {
  return {
    replaceState: function(){
      return true;
    },
    apolloProvider: null,
    i18n: null
  }
};
```

While using vuex that's a normal flow when store is initialized before all the boot files, but apollo cannot be initialized before the boot files. We need `router` and `ssrContext` arguments which are only can be accessed in the boot files. For this trick `store/index.js` used as a wrapper, and then it continues to initialize completely inside the boot file `apollo.js`.

One of the solutions might be to use different hydration variable (not `replaceState(window.__INITIAL_STATE__)`), or disabling store at all, but in that case the problem is the template rendering, which is not allowing to add `__INITIAL_STATE__` or other variable inside the template in the HEAD tag. Solution doesn't work completely fine, especially when starting to use cordova (it shows javascript data before the splashscreen).

Would be really great to be on the edge of quasar for our application, but this change is breaking our apollo integration. Please, lets find the solution to it.

One of the easiest solutions might be to hydrate data manually, instead of automatic way before the boot files. So would be great to have some flag, which can switch on/off the data hydration inside the `client-entry.js`.

**Comment by pdanpdan** [comment]
It looks like the normal workflow for vue-apollo is described here: https://vue-apollo.netlify.com/guide/ssr.html#create-apollo-client and here: https://vue-apollo.netlify.com/guide/ssr.html#create-app

And the app creation maps exactly on how Quasar works. The store is hydrated from window.__INITIAL_STATE__ and then in the boot file the apollo client cache is restored.

**Comment by rstoenescu** [comment]
Added a new Boolean flag in quasar.conf.js > ssr > manualHydration and will be available with "@quasar/app" v1.4.2+

**Comment by dmitry** [comment]
@pdanpdan yes, whole workflow worked nicely with quasar. https://github.com/quasarframework/quasar/commit/04657289e06a03309c9a0f9667bd72d214b701d4 solves an issue.
@rstoenescu thanks for the quick response!
